Modules/Forum/classes/class.ilObjForumNotificationDataProvider.php

```diff
@@ -67,18 +67,24 @@ class ilObjForumNotificationDataProvider implements ilForumNotificationMailData
 	 * @var bool
 	 */
 	protected $is_anonymized = false;
-	
+
+	/** @var ilForumNotificationCache */
+	private $notificationCache;
+
 	/**
 	 * @param ilForumPost $objPost
-	 * @param int         $ref_id
+	 * @param int $ref_id
+	 * @param ilForumNotificationCache $notificationCache
 	 */
-	public function __construct(ilForumPost $objPost, $ref_id)
+	public function __construct(ilForumPost $objPost, $ref_id, \ilForumNotificationCache $notificationCache)
 	{
 		global $DIC;
 		$this->db = $DIC->database();
 		$this->access = $DIC->access();
 		$this->user = $DIC->user();
-		
+
+		$this->notificationCache = new \ilForumNotificationCache();
```

Comment:
I don't get this :-). You already passed an instance of `\ilForumNotificationCache` to the constructor. Why creating a new instance here?

Reply:
You are completely right :+1: . Must be slipped through :see_no_evil: